src/test/java/com/aws/greengrass/componentmanager/ComponentServiceHelperTest.java

```diff
@@ -60,7 +62,6 @@ class ComponentServiceHelperTest {
     @BeforeEach
     void beforeEach() {
         PlatformResolver platformResolver = new PlatformResolver(null);
-        lenient().when(clientFactory.getGreengrassV2DataClient()).thenReturn(client);
```

Comment:
Would it be useful to have some tests still cover this code path and test `getGreengrassV2DataClient` ?

Reply:
Not sure which code paths you mean. No coverage was removed in this PR. I did notice that nothing in GreengrassServiceClientFactory is tested. So yeah, we should probably add those tests as a follow up.

Reply:
I meant that `getGreengrassV2DataClient()` is now replaced with `fetchGreengrassV2DataClient()` and tests are modified to use the new method. We could add some more tests that cover calls to `getGreengrassV2DataClient()`. But that can be a follow up.